app/service/govuk_site_lookup_service.rb

```diff
@@ -0,0 +1,29 @@
+class GovukSiteLookupService
+  def govuk_page?(slug)
+    public_url_info.key?(slug)
+  end
+
+  def page_link(slug)
+    public_url_info[slug][:link]
+  end
+
+  def page_title(slug)
+    public_url_info[slug][:title]
+  end
+
+  def public_url_info
+    Rails.cache.fetch("imminence_public_url_info", expires_in: 5.minutes) do
+      content_store_api = GdsApi.content_store
+      place_pages = GdsApi.search.search({ filter_format: "place", count: 200, fields: "title,link,content_id" })
+
+      public_url_info = {}
```

Comment:
Minor: we could use [each_with_object](https://apidock.com/rails/Enumerable/each_with_object) vs instantiating an empty hash and returning separately. E.g:

```
place_pages["results"].each_with_object do |pp, hsh|
  ci = content_store_api.content_item(pp["link"])
  slug = ci.to_hash["details"]["place_type"]
  hsh[slug] = { link: Plek.website_root + pp["link"], title: pp["title"] }
end
```

Reply:
Done!